### DISCUSSION TITLE
Folding arrow not showing when it should

### DISCUSSION THREAD

**Discussion by Lomic2** [atom/atom#22257] [Q&A]
Hi,

I really love coding with Atom, to me it is almost perfect. I particularly enjoy the folding tool, when the code is long and with a lot of parts, it is great.

But sometimes, for reasons I don’t get, some parts won’t fold, the arrow does not show up, like on this [image](https://nsa40.casimages.com/img/2021/04/22/2104221030166282.png).

That function “SelectStrength(strength)” is built like the functions after and before her, yet I can’t fold her. I can fold the ifs inside, but not the whole function.

This is all js, and the 1800+ lines and functions all fold neatly except that function.

I’ve read many threads about code folding before posting, I hope I didn’t miss one with the answer.

If someone has an idea, I am curious to find out.

Thanks !

**Comment by sadick254** [comment]
Hey @Lomic2 Thanks for bringing this up. This looks like a real issue that could be opened on https://github.com/atom/language-javascript. How can we reproduce this in the most minimal way? That would really help us identify where the issue is and hopefully resolve it.

**Comment by Lomic2** [comment]
Hi sadick254,

I would love to give more details but as I've continued working on my code, the issue disappeared and now I can fold that function like any other.

It is not the first time though, so if it happens again, I'll try to add more details. I'll look in my code if I have some part which doesn't fold inside one of my functions.

I was hoping that it was a known behaviour of Atom and that other users could have figured why sometimes you can't fold.

Thanks.

**Comment by Lomic2** [comment]
Oh I found one bit that doesn't fold : [image](https://nsa40.casimages.com/img/2021/04/22/210422060810430491.png)

Same exact "if" as just before, except some var names and object names but I don't see how those differences can affect the folding.

If I copy and paste those two ifs in a new js file, no folding issue anymore. Same if I copy the entire function, no more folding issues. Same with an empty function before and after that function.

**Reply by sadick254** [reply]
I would suggest you post the code as a snippet on your comment instead of attaching it as an image. That way it is more accessible and I can copy and paste to try and reproduce the folding issue you are experiencing. 
I don't think it is a known issue as I have not seen an issue resembling this reported anywhere within our repositories.

**Reply by Lomic2** [reply]
I know but when I copy/paste it myself in a blank page, the issue goes away so...
